src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,153 @@
 package core.basesyntax;
 
+import java.util.Arrays;
```

Comment:
The checklist advises against using utility classes like `Arrays`. Consider implementing your own method for copying arrays instead of using `Arrays.copyOf` .